### PR TITLE
Set reference kind when we know it should be set

### PR DESCRIPTION
**Pull request by tivervac**
see eclipse/xtext-core#965

### CODE REVIEW

**Comment by tivervac** [comment]
@cdietrich Might want to wait for master to build to see we didn't break anything with eclipse/xtext-core#965

File: org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeCrossrefProposalProvider.xtend [hunk 1]
```diff
@@ -44,7 +44,7 @@ class XbaseIdeCrossrefProposalProvider extends IdeCrossrefProposalProvider {
 					0
 			val converter = getTypeConverter(context.resource)
 			val objectOrProxy = candidate.getEObjectOrProxy
-			return proposalCreator.createProposal(proposalString, context) [ result |
+			val proposal = proposalCreator.createProposal(proposalString, context) [ result |
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeCrossrefProposalProvider.xtend]: Any reason not to move this to the closure?

Comment (cdietrich) [reply org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeCrossrefProposalProvider.xtend]: Am not sure if the create proposal can return null if prefix is bad

Comment (tivervac) [reply org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeCrossrefProposalProvider.xtend]: My fault, didn't notice that this was an initialisation closure. If I do it in the closure, it's nullsafe.
Will replace my current implementation for the one you proposed.

**Pull request merged**